src/main/java/org/opensearch/ad/rest/RestValidateDetectorAction.java

```diff
@@ -0,0 +1,44 @@
+package org.opensearch.ad.rest;
+
+import static org.opensearch.rest.RestRequest.Method.POST;
+import static org.opensearch.rest.RestStatus.NOT_FOUND;
+import static org.opensearch.rest.RestStatus.OK;
+
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ad.constant.CommonErrorMessages;
+import org.opensearch.ad.settings.EnabledSetting;
+import org.opensearch.extensions.rest.ExtensionRestRequest;
+import org.opensearch.extensions.rest.ExtensionRestResponse;
+import org.opensearch.rest.RestHandler.Route;
+import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.sdk.ExtensionRestHandler;
+
+public class RestValidateDetectorAction implements ExtensionRestHandler {
+    private final Logger logger = LogManager.getLogger(RestValidateDetectorAction.class);
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(POST, "/detectors"));
```

Comment:
Need the new routes here.

Reply:
Since Validate detector is used to return the detector configuration has any issue that prevent Opensearch from creating the detector, do you means we need add new route `Head` here for verifying that resource is available?

Reply:
Refer https://opensearch.org/docs/latest/monitoring-plugins/ad/api/#validate-detector for the route @mloufra.

Reply:
@mloufra there are 3 requests that will be handled, but only two routes: the wildcard `{type}` will handle two of them.  The specific routes you need to map are in the issue for [SDK #221](https://github.com/opensearch-project/opensearch-sdk-java/issues/221).

Reply:
@dbwiddis @owaiskazi19 Thank you for your comments. For my understand, Validate detector having two option of validation. First one is focus on the detector configuration to find any issues which will block detector creation. Second one is focus on the source data to check the probability of complete model training. So we need two routes `POST _plugins/_anomaly_detection/detectors/_validate/detector`, one for detector, and one for model `POST _plugins/_anomaly_detection/detectors/_validate/model`.

Reply:
Based on the issue we will 2 routes:
```
POST /detectors/_validate
POST /detectors/_validate/{type}

Reply:
> We are not covering the below route as a part of this issue
> 
> ```
> POST _plugins/_anomaly_detection/detectors/_validate/model
> ```

Actually, the `{type}` will include either `detector` or `model` that we'll extract with a `param.get("type")` as part of the logic later.

We could have listed the routes separately (`_validate/detector` and `_validate/model`) but using `{type}` parallels the existing plugin usage.